flexmeasures/data/models/planning/linear_optimization.py

```diff
@@ -171,6 +174,16 @@ def ems_derivative_min_select(m, j):
         else:
             return v
 
+    def device_efficiency(m, d, j):
+        """Assume perfect efficiency if no efficiency information is available."""
+        try:
+            eff = device_constraints[d]["efficiency"].iloc[j]
```

Comment:
We only seem to apply it to 0 - will we find it here for every `d`?
(Sorry if my mind is not wrapped correctly into this linear program after a while)

Reply:
Yes. So if we later add more flexible devices, their storage efficiency will get picked up here.

Reply:
When do we "later" do that?

Reply:
I'm talking about joined optimization of multiple (storage) devices. I guess that would be part of sector coupling, which is on our [roadmap](https://flexmeasures.io/roadmap/).

Reply:
Okay, fair enough. I believe I was under the belief that this code can still be used for a charging station, which might come even earlier on the roadmap.

Reply:
Yes, this code can currently be used by a charging station (or a standalone battery or a heat storage). It can also handle jointly scheduling multiple flexible storage devices, but the rest of our application does not support this yet.

But I don't think I understand your concern yet. Do my answers help, or am I missing your point?

Reply:
I thought we once supported multiple chargers (at one station) and still did.